### ISSUE TITLE
4.12+domains treats Bytes.unsafe_of_string / Bytes.unsafe_to_string as noop

### ISSUE THREAD

**Issue by hhugo** [ocamllabs/ocaml-multicore#668]
This is due to the following piece of code in `bytecomp/bytegen.ml`
```ocaml
  | Lprim((Pidentity | Pnop | Popaque |
           Pbytes_to_string | Pbytes_of_string), [arg], _) ->
      comp_expr env arg sz cont
```

It is not the case in ocaml.4.12 and seems to have been fixed in `ocaml-multicore#5.0`.

I found the issue while testing jsoo with multicore.4.12+domains. jsoo uses (--under a flag) a different memory representation for strings and bytes.

**Comment by kayceesrk** [comment]
@ctk21 we should backport the fix to `4.12+domains` and `4.12+domains+effects`.

**Comment by ctk21** [comment]
This is fixed in #675 and was backported to `4.12+domains`
